### PR TITLE
Fixes 4620 : Editing categories of an existing picture: Reuse categories selection UI from the Upload Wizard

### LINKED ISSUE #4620: Editing categories of an existing picture: Reuse categories selection UI from the Upload Wizard
I think categories editing should be revamped: It currently uses a special UI, which takes time to improve and maintain.

It would be better if it reused the existing category selection UI of the Upload Wizard.
adb

### PR DESCRIPTION
**Pull request by Ayan-10**
**Description (required)**

Fixes #4620

**What changes did you make and why?**

App currently uses a special UI, which takes time to improve and maintain. It would be better if it reused the existing category selection UI of the Upload Wizard.

- Reuse categories selection UI from the Upload Wizard
- Deleted old UI
- Added tests for new lines

**Tests performed (required)**

Tested latest prodDebug on Pixel 3 with API level 30.

**Screenshots (for UI changes only)**

**New UI**

**Old UI**

### CODE REVIEW

**Comment by Ayan-10** [comment]
@nicolas-raoul For now, I removed `values-zh-hk` directory so that the PR can be tested and reviewed. After approval of the PR, I can add the directory if required.

Thanks.

File: app/src/main/java/fr/free/nrw/commons/category/CategoryEditHelper.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import fr.free.nrw.commons.utils.ViewUtilWrapper;
 import io.reactivex.Observable;
 import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryEditHelper.java:17]: This line is not needed, right?

File: app/src/main/java/fr/free/nrw/commons/upload/categories/CategoriesPresenter.kt [hunk 2]
```diff
@@ -106,6 +144,88 @@ class CategoriesPresenter @Inject constructor(
...
+                            Timber.d("Categories are added.")
+                            media.addedCategories = selectedCategories
+                            repository.cleanup()
+                            view.dismissProgressDialog()
+                            view.refreshCategories()
+                            view.goBackToPreviousScreen()
+                        })
+                        {
+                            Timber.e(
+                                "Failed to update depictions"
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/categories/CategoriesPresenter.kt:220]: Do you mean `categories`?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
It works great, and the code looks fine to me.
Just two minor change requests.

File: app/src/main/java/fr/free/nrw/commons/upload/categories/CategoriesPresenter.kt (hunk 2 above)
Comment (Ayan-10) [reply app/src/main/java/fr/free/nrw/commons/upload/categories/CategoriesPresenter.kt]: Yes.

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**